Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -165,12 +172,27 @@ public void DeletionGracePeriodReminder3Sent()
     {
         return DeletionProcesses.FirstOrDefault(x => x.Status == deletionProcessStatus);
     }
+
+    public IdentityDeletionProcess CancelDeletionProcess(DeviceId canceledByDeviceId)
+    {
+        var deletionProcess = GetDeletionProcessInStatus(DeletionProcessStatus.Approved) ?? throw new DomainException(DomainErrors.NoDeletionProcessWithRequiredStatusExists());
+
+        if (deletionProcess.GracePeriodEndsAt < SystemTime.UtcNow)
```

Comment:
I would only check for the correct status. Once the grace period has expired, the deletion process will transition to the status `Deleting`.
In theory there is a short period of time were this transition didn't occur yet, even though the grace period is over. But we're okay with the deletion process being cancelled during this period.